runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -277,6 +278,7 @@ private ChannelInitializer getChannelInitializer() {
         return new ChannelInitializer() {
             @Override
             protected void initChannel(@Nonnull Channel ch) throws Exception {
+                ch.pipeline().addLast(new ReadTimeoutHandler(parameters.getHandshakeTimeout()));
```

Comment:
I see that we are using the handshake timeout here. But do we want handshake timeout and connection timeout be fundamentally the same? If yes, then this would fine, however if they may differ in the future we might want to introduce a new a corfu runtime parameter such as connection timeout.

Reply:
Better naming would be good. however my concern was about a scenario that we want to set-up different timeouts for inactivity and for handshake. It might be a contrived example but we might set an aggressive handshake time out of 10 sec (as it is now in the parameters) but want to allow a lenient timeout of 30 sec for inactivity.

Reply:
the timeout added to the pipeline by the handshake will be removed, if the handshake succeed. @annym this is correct right?

Reply:
Yes it will be removed. I think Sam is trying to say what if we want different values for the handshake timeout than for the inactiveConnectionTimeout. Also, the default value for the handshakeTimeout is 10 seconds, couldn't we end up closing the connection too frequently?

Reply:
addressed all comments. I think 30 seconds should suffice. Its configurable anyways.

Reply:
Thanks @Maithem.